### PR TITLE
Structure of package

### PR DESCRIPTION
**Pull request by Cxristian**
create runnnerCuckesTest and report

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -58,4 +65,4 @@ test {
     testLogging {
         events 'PASSED', 'STARTED', 'FAILED', 'SKIPPED'
     }
-}
+}
```
Comment (TheYisusGithub) [inline build.gradle]: Please dear friend put here a blank space.

File: src/main/java/com/foundation/salesforce/core/api/OpportunityApi.java [hunk 2]
```diff
@@ -46,10 +50,38 @@ public static OpportunityApi getInstance() {
         return new OpportunityApi();
     }
 
+    /**
+     * Add a deserialized json content to the body of 'request' Requestpecification attribute of this class.
+     *
+     * @param opportunityBody a flat deserialized String containing the key/value data to be provided to a POST or
```
Comment (TheYisusGithub) [inline src/main/java/com/foundation/salesforce/core/api/OpportunityApi.java]: Please do not exceed 120 lines.

File: src/main/java/com/foundation/salesforce/core/api/OpportunityApi.java [hunk 3]
```diff
@@ -46,10 +50,38 @@ public static OpportunityApi getInstance() {
...
+     *      PATCH request.
+     */
+    public void setContent(String opportunityBody) {
+        restClient.buildSpec(opportunityBody);
+    }
+
+    /**
+     * Add a json content to the body of 'request' Requestpecification attribute contained in a Map format.
+     *
+     * @param opportunityBody a Map structure containing the key/value data to be provided to a POST or PATCH request.
```
Comment (TheYisusGithub) [inline src/main/java/com/foundation/salesforce/core/api/OpportunityApi.java]: Please do not exceed 120 lines.

File: src/main/java/com/foundation/salesforce/core/api/OpportunityApi.java [hunk 4]
```diff
@@ -46,10 +50,38 @@ public static OpportunityApi getInstance() {
...
+     */
+    public void setContent(Map opportunityBody) {
+        restClient.buildSpec(opportunityBody);
+    }
+
+    /**
+     * Adds content to the body of 'request' Requestpecification attribute contained within a JSONObject.
+     *
+     * @param opportunityBody a JSONObject containing the key/value data to be provided to a POST or PATCH request.
```
Comment (TheYisusGithub) [inline src/main/java/com/foundation/salesforce/core/api/OpportunityApi.java]: Please do not exceed 120 lines.

File: src/main/java/com/foundation/salesforce/core/report/GeneratorReport.java [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+import net.masterthought.cucumber.Configuration;
+import net.masterthought.cucumber.ReportBuilder;
+import net.masterthought.cucumber.Reportable;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * GeneratorReport class
```
Comment (TheYisusGithub) [inline src/main/java/com/foundation/salesforce/core/report/GeneratorReport.java]: Please you can be more detail for this class explanations.

File: build.gradle (hunk 1 above)
Comment (titor-oopart) [inline build.gradle]: add an empty line

File: src/main/java/com/foundation/salesforce/core/report/GeneratorReport.java [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+ *
+ * @author Cristian Lujan
```
Comment (TheYisusGithub) [inline src/main/java/com/foundation/salesforce/core/report/GeneratorReport.java:26]: Please put point at the end to the line.

File: src/main/java/com/foundation/salesforce/core/report/GeneratorReport.java [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+        final boolean runWithJenkins = false;
+
+        final Configuration configuration = new Configuration(reportOutputDirectory, projectName);
+        // additional metadata presented on main page
+        configuration.addClassifications("Branch", "RELEASE/1.0");
+        final ReportBuilder reportBuilder = new ReportBuilder(jsonFiles, configuration);
+        final Reportable result = reportBuilder.generateReports();
+        // and here validate 'result' to decide what to do if report has failed
+    }
+}
```
Comment (titor-oopart) [inline src/main/java/com/foundation/salesforce/core/report/GeneratorReport.java]: add an empty line

File: src/main/java/com/foundation/salesforce/core/report/GeneratorReport.java [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * GeneratorReport class
+ *
+ * @author Cristian Lujan
+ * @version 1.0
```
Comment (TheYisusGithub) [inline src/main/java/com/foundation/salesforce/core/report/GeneratorReport.java:27]: Please put point at the end to the line.

File: src/test/java/com/foundation/salesforce/Hooks/Hooks.java [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+
+    public Hooks(Context context) {
+        this.context = context;
+    }
+
+    @After("@deleteNewWorkspace")
+    public void deleteNewWorkspace() {
+        OpportunityApi.getInstance().deleteOpportunityById(context.getOpportunity().getId());
+    }
+}
```
Comment (titor-oopart) [inline src/test/java/com/foundation/salesforce/Hooks/Hooks.java]: add an empty line

File: src/test/java/com/foundation/salesforce/runner/RunCukesTest.java [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+public class RunCukesTest extends AbstractTestNGCucumberTests {
+
+    /**
+     * The method afterExecution executes the project and generate it report.
+     */
+    @AfterTest
+    public void afterExecution() {
+        GeneratorReport.getInstance().generateReport();
+    }
+}
```
Comment (titor-oopart) [inline src/test/java/com/foundation/salesforce/runner/RunCukesTest.java]: add an empty line

File: src/test/java/com/foundation/salesforce/steps/OpportunitySteps.java [hunk 11]
```diff
@@ -0,0 +1,66 @@
...
+        opportunityApi.setContent(inputBody);
+    }
+
+    @When("User send de request post to opportunity endpoint")
+    public void SendDeRequestPostToOpportunityEndpoint() {
+        this.response = opportunityApi.createOpportunity();
+        opportunity.setId(response.jsonPath().getString("id"));
+    }
+
+}
```
Comment (titor-oopart) [inline src/test/java/com/foundation/salesforce/steps/OpportunitySteps.java]: add an empty line

File: src/test/java/com/foundation/salesforce/Hooks/Hooks.java [hunk 12]
```diff
@@ -0,0 +1,26 @@
+package com.foundation.salesforce.Hooks;
```
Comment (MelissaRoman06) [inline src/test/java/com/foundation/salesforce/Hooks/Hooks.java]: Please add copyright.

File: src/test/java/com/foundation/salesforce/runner/RunCukesTest.java [hunk 13]
```diff
@@ -0,0 +1,44 @@
...
+package com.foundation.salesforce.runner;
+
+import com.foundation.salesforce.core.report.GeneratorReport;
+import cucumber.api.CucumberOptions;
+import cucumber.api.testng.AbstractTestNGCucumberTests;
+import org.testng.annotations.AfterTest;
+
+
+/**
+ * RunCukesTest class
```
Comment (TheYisusGithub) [inline src/test/java/com/foundation/salesforce/runner/RunCukesTest.java]: Please you most be more descriptive.

File: src/test/java/com/foundation/salesforce/steps/OpportunitySteps.java [hunk 14]
```diff
@@ -0,0 +1,66 @@
...
+import com.foundation.salesforce.entities.Context;
+import com.foundation.salesforce.entities.Opportunity;
+import cucumber.api.java.en.Given;
+import cucumber.api.java.en.When;
+import io.restassured.response.Response;
+
+import java.util.Map;
+
+/**
+ * OpportunitySteps class
```
Comment (TheYisusGithub) [inline src/test/java/com/foundation/salesforce/steps/OpportunitySteps.java]: Please you can be more descreptive for your class.

File: src/test/java/com/foundation/salesforce/steps/OpportunitySteps.java [hunk 15]
```diff
@@ -0,0 +1,66 @@
...
+ *
+ * @author Cristian Lujan
```
Comment (TheYisusGithub) [inline src/test/java/com/foundation/salesforce/steps/OpportunitySteps.java:28]: Please put point at the that your name.

File: src/test/java/com/foundation/salesforce/steps/OpportunitySteps.java [hunk 16]
```diff
@@ -0,0 +1,66 @@
...
+ * @version 1.0
```
Comment (TheYisusGithub) [inline src/test/java/com/foundation/salesforce/steps/OpportunitySteps.java:29]: Please put point at the end to the version.

File: src/test/java/com/foundation/salesforce/runner/RunCukesTest.java [hunk 17]
```diff
@@ -0,0 +1,44 @@
...
+ *
+ * @author Cristian Lujan
```
Comment (TheYisusGithub) [inline src/test/java/com/foundation/salesforce/runner/RunCukesTest.java:24]: Please put point at the end your name.

File: src/test/java/com/foundation/salesforce/runner/RunCukesTest.java [hunk 18]
```diff
@@ -0,0 +1,44 @@
...
+ * @version 1.0
```
Comment (TheYisusGithub) [inline src/test/java/com/foundation/salesforce/runner/RunCukesTest.java:25]: Please put point at the to your version.

File: src/test/resources/features/opportunity.feature [hunk 19]
```diff
@@ -0,0 +1,13 @@
+
+Feature: Manage Opportunities tests
```
Comment (TheYisusGithub) [inline src/test/resources/features/opportunity.feature:2]: I think you would be more description in your features.

File: src/test/java/com/foundation/salesforce/steps/OpportunitySteps.java [hunk 20]
```diff
@@ -0,0 +1,65 @@
...
+import com.foundation.salesforce.entities.Context;
+import com.foundation.salesforce.entities.Opportunity;
+import cucumber.api.java.en.Given;
+import cucumber.api.java.en.When;
+import io.restassured.response.Response;
+
+import java.util.Map;
+
+/**
+ * OpportunitySteps class.
```
Comment (TheYisusGithub) [inline src/test/java/com/foundation/salesforce/steps/OpportunitySteps.java:26]: Please you most be more descriptive in your class.

File: src/test/java/com/foundation/salesforce/runner/RunCukesTest.java (hunk 18 above)
Comment (alszla) [reply src/test/java/com/foundation/salesforce/runner/RunCukesTest.java:25]: Erhmmm, AFAIK I guess there's no need to place a dot at the the of @version

**Review by alszla — APPROVED** [review]

File: src/test/java/com/foundation/salesforce/Hooks/Hooks.java (hunk 12 above)
Comment (Cxristian) [reply src/test/java/com/foundation/salesforce/Hooks/Hooks.java]: fixed

File: src/test/java/com/foundation/salesforce/steps/OpportunitySteps.java [hunk 21]
```diff
@@ -0,0 +1,65 @@
...
+    /**
+     * Constructor of contact com.foundation.salesforce.steps sending the context.
+     *
+     * @param context init the context.
+     */
+    public OpportunitySteps(final Context context) {
+        this.context = context;
+        opportunity = context.getOpportunity();
+    }
+
```
Comment (titor-oopart) [inline src/test/java/com/foundation/salesforce/steps/OpportunitySteps.java:48]: add documentation

File: build.gradle (hunk 1 above)
Comment (Cxristian) [reply build.gradle]: fixed

File: src/main/java/com/foundation/salesforce/core/report/GeneratorReport.java (hunk 6 above)
Comment (Cxristian) [reply src/main/java/com/foundation/salesforce/core/report/GeneratorReport.java:26]: fixed

File: src/main/java/com/foundation/salesforce/core/report/GeneratorReport.java (hunk 8 above)
Comment (Cxristian) [reply src/main/java/com/foundation/salesforce/core/report/GeneratorReport.java:27]: fixed

File: build.gradle (hunk 1 above)
Comment (Cxristian) [reply build.gradle]: fixed

File: src/main/java/com/foundation/salesforce/core/report/GeneratorReport.java (hunk 7 above)
Comment (Cxristian) [reply src/main/java/com/foundation/salesforce/core/report/GeneratorReport.java]: fixed

File: src/test/java/com/foundation/salesforce/Hooks/Hooks.java (hunk 9 above)
Comment (Cxristian) [reply src/test/java/com/foundation/salesforce/Hooks/Hooks.java]: fixed

File: src/test/java/com/foundation/salesforce/runner/RunCukesTest.java (hunk 10 above)
Comment (Cxristian) [reply src/test/java/com/foundation/salesforce/runner/RunCukesTest.java]: fixed

File: src/test/java/com/foundation/salesforce/steps/OpportunitySteps.java (hunk 11 above)
Comment (Cxristian) [reply src/test/java/com/foundation/salesforce/steps/OpportunitySteps.java]: fixed

File: src/test/java/com/foundation/salesforce/steps/OpportunitySteps.java (hunk 16 above)
Comment (Cxristian) [reply src/test/java/com/foundation/salesforce/steps/OpportunitySteps.java:29]: the version has no dot...

**Review by TheYisusGithub — APPROVED** [review]

**Review by titor-oopart — APPROVED** [review]

File: src/test/java/com/foundation/salesforce/steps/OpportunitySteps.java (hunk 15 above)
Comment (MelissaRoman06) [reply src/test/java/com/foundation/salesforce/steps/OpportunitySteps.java:28]: That is not necessary according to our code convention.

File: src/test/java/com/foundation/salesforce/steps/OpportunitySteps.java [hunk 22]
```diff
@@ -0,0 +1,65 @@
...
+    @Given("User log in into the Opportunities page")
+    public void LogInIntoTheOpportunitiesPage() {
+        opportunityApi.getInstance().getOpportunity();
+    }
+
+
+    @Given("User set up the data:")
```
Comment (MelissaRoman06) [inline src/test/java/com/foundation/salesforce/steps/OpportunitySteps.java:55]: Please add documentation to every method.

**Pull request merged**